packages/components/pie-button/src/index.ts

```diff
@@ -157,17 +173,17 @@ export class PieButton extends FormControlMixin(LitElement) implements ButtonPro
     }
 
     private _handleClick () {
-        if (!this.isLoading && this.form) {
-            if (this.type === 'submit') {
-                // only submit the form if either formnovalidate is set, or the form passes validation checks (triggers native form validation)
-                if (this.formnovalidate || this.form.reportValidity()) {
-                    this._simulateNativeButtonClick('submit');
-                }
-            }
+        if (!this.form) return;
+        if (this.isLoading) return;
+        if (this.tag !== 'button') return;
```

Comment:
absolutely just a preference but I feel like this tag check should be first (or loading)